x-pack/plugins/cloud_security_posture/public/common/api/use_compliance_dashboard_data_api.ts

```diff
@@ -7,14 +7,36 @@
 
 import { type QueryObserverOptions, useQuery } from '@tanstack/react-query';
 import { useKibana } from '../hooks/use_kibana';
-import { ComplianceDashboardData } from '../../../common/types';
+import { PosturePolicyTemplate, ComplianceDashboardData } from '../../../common/types';
 import { STATS_ROUTE_PATH } from '../../../common/constants';
 
-const getStatsKey = ['csp_dashboard_stats'];
+const getCspmStatsKey = ['csp_cspm_dashboard_stats'];
+const getKspmStatsKey = ['csp_kspm_dashboard_stats'];
 
-export const useComplianceDashboardDataApi = (
+export const getStatsRoute = (policyTemplate: PosturePolicyTemplate) => {
+  return STATS_ROUTE_PATH.replace('{policy_template}', policyTemplate);
+};
+
+export const useCspmComplianceDashboardDataApi = (
```

Comment:
_nit_
consider renaming to smth like `useStatsCSPM` / `useStatsKSPM`

to me `ComplianceDashboard` says where it's used but not what it does, and i also think `DataApi` is not needed as i think any `use-query` hook will return data while managing the async nature of how we got it. (we need to access `query.data` to get the result)